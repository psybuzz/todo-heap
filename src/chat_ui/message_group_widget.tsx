import * as React from 'react';

import { Message } from 'semantic-ui-react'

export const MessageGroupWidget = () => (
  <Message>
    <Message.Header>list -n 5</Message.Header>
    <Message.List>
      <Message.Item>add feedback to the contribution guide</Message.Item>
      <Message.Item>check the HTTP status updates</Message.Item>
      <Message.Item>add feedback to the contribution guide</Message.Item>
      <Message.Item>check the HTTP status updates</Message.Item>
      <Message.Item>add feedback to the contribution guide</Message.Item>
      <Message.Item>check the HTTP status updates</Message.Item>
    </Message.List>
  </Message>
)
