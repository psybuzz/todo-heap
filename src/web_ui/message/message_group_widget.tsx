import * as React from 'react';

import { Message } from 'semantic-ui-react';

import {MessageItemWidget, MessageItemProps} from './';

export interface MessageGroupProps {
  messages: MessageItemProps[];
  id: string;
  command: string;
  warnings?: string[];
}

export interface MessageWarningProps {
  text: string;
}

export const MessageWarningWidget = (props: MessageWarningProps) => (
  <Message error>
    <Message.Item>
      {props.text}
    </Message.Item>
  </Message>
);

export const MessageGroupWidget = (props: MessageGroupProps) => {
  const warnings = props.warnings ? props.warnings.map(w =>
            <MessageWarningWidget text={w}>
            </MessageWarningWidget>
          ): null;
  return (
    <Message>
      <Message.Header>{props.command}</Message.Header>
      <Message.List>
        {warnings}
        {props.messages.map(m =>
          <MessageItemWidget
            todoRef={m.todoRef}
            key={m.id} id={m.id}
            text={m.text}
            priority={m.priority}/>)}
      </Message.List>
    </Message>
  );
};