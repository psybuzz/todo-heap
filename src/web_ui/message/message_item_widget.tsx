import * as React from 'react';

import { Message } from 'semantic-ui-react'

export interface MessageItemProps {
  priority: number,
  text: string,
  id: string,
}

export const MessageItemWidget = (props: MessageItemProps) => (
  <Message.Item>P{props.priority} {props.text}</Message.Item>
)