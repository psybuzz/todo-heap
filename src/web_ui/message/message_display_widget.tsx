import * as React from 'react';

import {MessageGroupWidget, MessageGroupProps} from './message_group_widget';

export type MessageGroups = MessageGroupProps[];
export interface MessageDisplayProps {
  groups: MessageGroups;
}

export const MessageDisplayWidget = (props: MessageDisplayProps) => (
  <div id='main'>
    {props.groups.map(group => <MessageGroupWidget
      key={group.id} id={group.id}
      command={group.command}
      messages={group.messages}/>)}
  </div>
)
