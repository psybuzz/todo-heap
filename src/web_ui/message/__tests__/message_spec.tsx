import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import * as renderer from 'react-test-renderer';

import {MessageItemWidget} from '../message_item_widget';
import {MessageGroupWidget, MessageGroupProps} from '../message_group_widget';
import {MessageDisplayWidget} from '../message_display_widget';

it('renders a message item', () => {
  const wrapper = mount(<MessageItemWidget id={'1'} key={'1'} priority={1} text={'Take out the trash'}/>);
  expect(wrapper.text()).toEqual("P1 Take out the trash");
});

it('renders a message group', () => {
  const messages = [
    {id: '1', priority: 1, text: 'foo bar'},
    {id: '2', priority: 1, text: 'baz'},
    {id: '3', priority: 2, text: 'what'},
  ];

  const component = renderer.create(
    <MessageGroupWidget messages={messages} id={'1'} command='ls -n 5'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a message display', () => {
  const messages = [
    {id: '1', priority: 1, text: 'foo bar'},
    {id: '2', priority: 1, text: 'baz'},
    {id: '3', priority: 2, text: 'what'},
  ];
  const groups = [
    {id: '1', command: 'list -n 10', messages},
    {id: '2', command: 'list -n 10', messages},
    {id: '3', command: 'list -n 10', messages},
  ];

  const component = renderer.create(
    <MessageDisplayWidget groups={groups}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});