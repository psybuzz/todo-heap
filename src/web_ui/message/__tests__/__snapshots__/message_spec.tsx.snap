exports[`test renders a message display 1`] = `
<div
  id="main">
  <div
    className="ui message">
    <div
      className="header">
      list -n 10
    </div>
    <ul
      className="list">
      <li
        className="content">
        P
        1
         
        foo bar
      </li>
      <li
        className="content">
        P
        1
         
        baz
      </li>
      <li
        className="content">
        P
        2
         
        what
      </li>
    </ul>
  </div>
  <div
    className="ui message">
    <div
      className="header">
      list -n 10
    </div>
    <ul
      className="list">
      <li
        className="content">
        P
        1
         
        foo bar
      </li>
      <li
        className="content">
        P
        1
         
        baz
      </li>
      <li
        className="content">
        P
        2
         
        what
      </li>
    </ul>
  </div>
  <div
    className="ui message">
    <div
      className="header">
      list -n 10
    </div>
    <ul
      className="list">
      <li
        className="content">
        P
        1
         
        foo bar
      </li>
      <li
        className="content">
        P
        1
         
        baz
      </li>
      <li
        className="content">
        P
        2
         
        what
      </li>
    </ul>
  </div>
</div>
`;

exports[`test renders a message group 1`] = `
<div
  className="ui message">
  <div
    className="header">
    ls -n 5
  </div>
  <ul
    className="list">
    <li
      className="content">
      P
      1
       
      foo bar
    </li>
    <li
      className="content">
      P
      1
       
      baz
    </li>
    <li
      className="content">
      P
      2
       
      what
    </li>
  </ul>
</div>
`;
